include/laser_filters/angular_bounds_filter.h

```diff
@@ -79,7 +79,7 @@ namespace laser_filters
           if(start_angle < lower_angle_){
             start_angle += input_scan.angle_increment;
             current_angle += input_scan.angle_increment;
-            start_time.set__sec(start_time.sec + input_scan.time_increment);
+            start_time.set__nanosec(start_time.nanosec + (input_scan.time_increment * 10e9)); // convert time increment to nanoseconds
```

Comment:
I think there are still a couple of issues:
- there are 1e9 nanoseconds in a second, so you should multiply by 1e9, not 10e9, right?
- set_nanosec only updates the number of nanoseconds.  The Time message type only allows nanosec to be between 0 and 1e9, so if the scan starts near the end of a second and we remove a bunch of beams from the beginning, then we could end up with an invalid number of nanoseconds. Instead we would want to add `nanoseconds / 1e9` to start_time.sec and `nanoseconds % 1e9` to start_time.nanosec

Time math is always tricky to get just right, so if there's a way to do this using utility functions that handle the details for us, that would be ideal.  The duration type should handle converting a floating point number of seconds to a proper duration for us in its from_seconds method: https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Duration.html#a39ce5aa6d0bb190d7d59857aa7a9af1a

How about:
```
start_time += Duration::from_seconds(input_scan.time_increment);
```

Reply:
Thank you for your review and pointing out the remaining issues.
You're right, it should be 1e9.
I will try to use the time duration as you recommend.

Reply:
Sorry, was on holliday and didn't had the hardware for testing. But the start time is now calculated with the Duration object :smile: 
I tried with the rplidar which publishes:
```
angle_min: -3.1241390705108643
angle_max: 3.1415927410125732
```
And the settings for the filter:
```
scan_to_scan_filter_chain:
  ros__parameters:
    filter1:
      name: angle
      type: laser_filters/LaserScanAngularBoundsFilter
      params:
        lower_angle: -1.52
        upper_angle: 1.52  
```
![2023-10-30_17-32](https://github.com/ros-perception/laser_filters/assets/61586098/700d2d1a-e8f5-48b0-a81c-71fd0786a3cb)
In the second case I tried to cut off more data at the beginning. The start time of the filtered data (red) is slightly more delayed.
```
scan_to_scan_filter_chain:
  ros__parameters:
    filter1:
      name: angle
      type: laser_filters/LaserScanAngularBoundsFilter
      params:
        lower_angle: 1.5
        upper_angle: 1.52  
```
![2023-10-30_17-33](https://github.com/ros-perception/laser_filters/assets/61586098/b5c0b6dc-73f2-462f-8ea8-5c3746b6b77f)
It looks correct to me. But please proof it.